framework/src/main/java/org/tron/core/services/event/BlockEventGet.java

```diff
@@ -88,18 +94,13 @@ public BlockEvent getBlockEvent(long blockNum) throws Exception {
   }
 
   public SmartContractTrigger getContractTrigger(BlockCapsule block, long solidNum) {
-    TransactionRetCapsule result;
-    try {
-      result = manager.getChainBaseManager().getTransactionRetStore()
-        .getTransactionInfoByBlockNum(ByteArray.fromLong(block.getNum()));
-    } catch (BadItemException e) {
-      throw new RuntimeException(e);
-    }
+
+    GrpcAPI.TransactionInfoList list = manager.getTransactionInfoByBlockNum(block.getNum());
 
     SmartContractTrigger contractTrigger = new SmartContractTrigger();
     for (int i = 0; i < block.getTransactions().size(); i++) {
       Protocol.Transaction tx = block.getInstance().getTransactions(i);
-      Protocol.TransactionInfo txInfo = result.getInstance().getTransactioninfo(i);
+      Protocol.TransactionInfo txInfo = list.getTransactionInfo(i);
```

Comment:
if list get from getTransactionInfoByBlockNum is empty, or list.getTransactionInfoCount() != block.getTransactions().size(), is there any problem?